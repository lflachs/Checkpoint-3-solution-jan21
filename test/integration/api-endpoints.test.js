const app = require('../../app');
const agent = require('supertest').agent(app);
const client = require('../../config/prismaClient');
const packageJson = require('../../package.json');

const createError = require('http-errors');

afterAll(async () => {
	await client.$disconnect();
});
describe('API Endpoints', () => {
	it('GET / should return the name of the project and the version', async () => {
		const response = await agent
			.get('/')
			.expect(200)
			.expect('content-type', /json/);
		expect(response.body).toMatchSnapshot();
	});
	it('GET /404 should return a status 404 with not found', async () => {
		const response = await agent
			.get('/404')
			.expect(404)
			.expect('content-type', /json/);
		expect(response.body).toBeDefined();
		expect(response.body.message).toBe(createError(404).message);
	});
});
