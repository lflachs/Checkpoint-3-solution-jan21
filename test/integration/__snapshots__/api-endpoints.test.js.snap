// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Endpoints GET / should return the name of the project and the version 1`] = `
Object {
  "name": "checkpoint-solution-jan21",
  "version": "1.0.0",
}
`;
